public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -31,62 +32,33 @@ export default class CloudMonitoringDatasource extends DataSourceApi<CloudMonito
     super(instanceSettings);
     this.authenticationType = instanceSettings.jsonData.authenticationType || 'jwt';
     this.api = new API(`${instanceSettings.url!}/cloudmonitoring/v3/projects/`);
-
     this.variables = new CloudMonitoringVariableSupport(this);
   }
 
   getVariables() {
     return this.templateSrv.getVariables().map(v => `$${v.name}`);
   }
 
-  query(options: DataQueryRequest<CloudMonitoringQuery>): Observable<DataQueryResponseData> {
-    return this.getTimeSeries(options).pipe(
-      map(data => {
-        if (!data.results) {
-          return { data: [] };
-        }
-
-        const result: DataQueryResponseData[] = [];
-        const values = Object.values(data.results);
-        for (const queryRes of values) {
-          if (!queryRes.series) {
-            continue;
-          }
-
-          const unit = this.resolvePanelUnitFromTargets(options.targets);
-
-          for (const series of queryRes.series) {
-            let timeSerie: any = {
-              target: series.name,
-              datapoints: series.points,
-              refId: queryRes.refId,
-              meta: queryRes.meta,
-            };
-            if (unit) {
-              timeSerie = { ...timeSerie, unit };
-            }
-            const df = toDataFrame(timeSerie);
-
-            for (const field of df.fields) {
-              if (queryRes.meta?.deepLink && queryRes.meta?.deepLink.length > 0) {
-                field.config.links = [
-                  {
-                    url: queryRes.meta?.deepLink,
-                    title: 'View in Metrics Explorer',
-                    targetBlank: true,
-                  },
-                ];
-              }
-            }
-            result.push(df);
-          }
-        }
-
-        return { data: result };
+  query(request: DataQueryRequest<CloudMonitoringQuery>): Observable<DataQueryResponse> {
+    this.intervalMs = request.intervalMs;
+    request.targets = request.targets.map(this.migrateQuery);
+    return super.query(request).pipe(
+      mergeMap((res: DataQueryResponse) => {
+        return from(this.processResponse(request, res));
```

Comment:
This should be simplified to:
```suggestion
    return super.query(request);
```